worlds/sc2wol/PoolFilter.py

```diff
@@ -18,46 +20,59 @@
 FACTORY_UNITS = {"Hellion", "Vulture", "Goliath", "Diamondback", "Siege Tank", "Thor", "Predator", "Widow Mine"}
 STARPORT_UNITS = {"Medivac", "Wraith", "Viking", "Banshee", "Battlecruiser", "Hercules", "Science Vessel", "Raven", "Liberator", "Valkyrie"}
 
-PROTOSS_REGIONS = {"A Sinister Turn", "Echoes of the Future", "In Utter Darkness"}
 
-
-def filter_missions(multiworld: MultiWorld, player: int) -> Dict[int, List[str]]:
+def filter_missions(multiworld: MultiWorld, player: int) -> Dict[MissionPools, List[str]]:
     """
     Returns a semi-randomly pruned tuple of no-build, easy, medium, and hard mission sets
     """
 
     mission_order_type = get_option_value(multiworld, player, "mission_order")
     shuffle_no_build = get_option_value(multiworld, player, "shuffle_no_build")
-    shuffle_protoss = get_option_value(multiworld, player, "shuffle_protoss")
+    enabled_campaigns = get_enabled_campaigns(multiworld, player)
+    disabled_campaigns = get_disabled_campaigns(multiworld, player)
     excluded_missions = get_option_value(multiworld, player, "excluded_missions")
-    final_map = get_option_value(multiworld, player, "final_map")
     mission_pools = {
-        MissionPools.STARTER: no_build_regions_list[:],
-        MissionPools.EASY: easy_regions_list[:],
-        MissionPools.MEDIUM: medium_regions_list[:],
-        MissionPools.HARD: hard_regions_list[:],
+        MissionPools.STARTER: [mission.mission_name for mission in get_missions_by_pool(MissionPools.STARTER)],
+        MissionPools.EASY: [mission.mission_name for mission in get_missions_by_pool(MissionPools.EASY)],
+        MissionPools.MEDIUM: [mission.mission_name for mission in get_missions_by_pool(MissionPools.MEDIUM)],
+        MissionPools.HARD: [mission.mission_name for mission in get_missions_by_pool(MissionPools.HARD)],
+        MissionPools.VERY_HARD: [mission.mission_name for mission in get_missions_by_pool(MissionPools.VERY_HARD)],
```

Comment:
`get_missions_by_pool` is a static operation for each pool, so iterating the `SC2Mission` enum five times is unnecessary. I suggest either something like this:
```
for mission in SC2Mission:
    mission_pools[mission.pool].append(mission.name)
```
... right here in `filter_missions` to do the work once per SC2 world, or doing it in `MissionTables`, which would do it once per generation if I remember AP's world handling correctly, and then `mission_pools` can just copy that work once per world.

Reply:
Updated